stream-chat-android-ui-components/src/test/kotlin/io/getstream/chat/android/ui/viewmodels/ChannelListViewModelTest.kt

```diff
@@ -0,0 +1,335 @@
+                channelsStateData = ChannelsStateData.Result(listOf(channel1, channel2)),
+                loading = false
+            )
+            .givenChannelMutes()
+            .givenLeaveChannel(listOf(user.id))
+            .get()
+
+        viewModel.leaveChannel(channel1)
+
+        verify(channelClient).removeMembers(listOf(user.id))
+    }
+
+    @Test
+    fun `Given channel list in content state When deleting a channel Should delete the channel`() = runTest {
+        val chatClient: ChatClient = mock()
+        val channelClient: ChannelClient = mock()
+
+        val viewModel = Fixture(chatClient, channelClient)
+            .givenCurrentUser()
+            .givenChannelsQuery()
+            .givenChannelsState(
+                channelsStateData = ChannelsStateData.Result(listOf(channel1, channel2)),
+                loading = false
+            )
+            .givenChannelMutes()
+            .givenDeleteChannel()
+            .get()
+
+        viewModel.deleteChannel(channel1)
+
+        verify(channelClient).delete()
+    }
+
+    @Test
+    fun `Given channel list in content state When hiding a channel Should hide the channel`() = runTest {
+        val chatClient: ChatClient = mock()
+        val channelClient: ChannelClient = mock()
+
+        val viewModel = Fixture(chatClient, channelClient)
+            .givenCurrentUser()
+            .givenChannelsQuery()
+            .givenChannelsState(
+                channelsStateData = ChannelsStateData.Result(listOf(channel1, channel2)),
+                loading = false
+            )
+            .givenChannelMutes()
+            .givenHideChannel()
+            .get()
+
+        viewModel.hideChannel(channel1)
+
+        verify(chatClient).hideChannel(
+            channelType = channel1.type,
+            channelId = channel1.id
+        )
+    }
+
+    @Test
+    fun `Given channel list in content state and the current user is online When loading more channels Should load more channels`() =
+        runTest {
+            val nextPageRequest = QueryChannelsRequest(
+                filter = queryFilter,
+                querySort = querySort,
+                offset = 30,
+                limit = 30,
+            )
+
+            val chatClient: ChatClient = mock()
+
+            val viewModel = Fixture(chatClient)
+                .givenCurrentUser()
+                .givenChannelsQuery()
+                .givenChannelsState(
+                    channelsStateData = ChannelsStateData.Result(listOf(channel1, channel2)),
+                    nextPageRequest = nextPageRequest,
+                    loading = false
+                )
+                .givenChannelMutes()
+                .givenIsOffline(false)
+                .get()
+
+            val mockObserver: Observer<ChannelListViewModel.State> = mock()
+            viewModel.state.observeForever(mockObserver)
+
+            viewModel.onAction(ChannelListViewModel.Action.ReachedEndOfList)
+
```

Comment:
What do you think about adding `removeObserver` with some other verifying below?

```kotlin
val state = ChannelListViewModel.State(isLoading = false, channels = listOf(channel1, channel2))
verify(mockObserver, atLeastOnce()).onChanged(state)
viewModel.state.removeObserver(mockObserver)
```

Reply:
Good suggestion.

Removed the observer at the end of the test.